### PR TITLE
feat: Added HotelInformation

### PR DESCRIPTION
**Pull request by kristianbinau**
## Description

- Added HotelInformation
## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Comment by kristianbinau** [comment]
## Known problem:
Since "Information" doesn't have a plural form, it can be difficult to fit into our predefined structure.. I decided that it would make sense to still keep the "s" in some places to make it clearer that there's more than one `HotelInformation`.

File: src/modules/hotel/__test__/getHotelInformationsByHotel.test.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
+import { FastifyInstance } from "fastify";
+import { build } from "../../../index";
+import { prisma } from "../../../plugins/prisma";
+
+describe("GET /api/hotel/:id/information", () => {
+    let fastify: FastifyInstance;
+
+    beforeAll(async () => {
+        fastify = await build();
+    });
```
Comment (frederikpyt) [inline src/modules/hotel/__test__/getHotelInformationsByHotel.test.ts:10]:
```suggestion
import { prisma } from "../../../plugins/prisma";

describe("GET /api/hotel/:id/information", () => {
    const fastify = gloabl.fastify;
```

File: src/modules/hotel/__test__/getHotelInformationsByHotel.test.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                key: 'Wifi Password',
+                value: 'overnites-is-cool',
+                hotelId: 1001,
+            },
+        });
+    });
+
+    afterAll(async () => {
+        await fastify.close();
+    });
```
Comment (frederikpyt) [inline src/modules/hotel/__test__/getHotelInformationsByHotel.test.ts:60]:
```suggestion
```

File: src/modules/hotel/hotel.schema.ts [hunk 3]
```diff
@@ -65,6 +66,12 @@ const getHotelSettingsByHotelSchema = z.object({
 
 const getHotelSettingsByHotelResponseSchema = z.array(hotelSettingSchema);
 
+const getHotelInformationsByHotelSchema = z.object({
+    id: z.string(),
```
Comment (frederikpyt) [inline src/modules/hotel/hotel.schema.ts:70]:
```suggestion
    id: z.number(),
```

File: src/modules/hotel/hotel.schema.ts [hunk 4]
```diff
@@ -65,6 +66,12 @@ const getHotelSettingsByHotelSchema = z.object({
...
+});
+
+const getHotelInformationsByHotelResponseSchema = z.array(hotelInformationSchema);
+
 const getFloorsByHotelSchema = z.object({
     id: z.string(),
```
Comment (frederikpyt) [inline src/modules/hotel/hotel.schema.ts:76]: z.number ??

File: src/modules/hotel/information/__test__/createHotelInformation.test.ts [hunk 5]
```diff
@@ -0,0 +1,183 @@
+import { User, Permission } from "@prisma/client";
+import { FastifyInstance } from "fastify";
+import { build } from "../../../../index";
+import { prisma } from "../../../../plugins/prisma";
+import { addTestUserAndPermission } from "../../../../utils/testHelper";
+
+describe("POST /api/hotel/information", () => {
+    let fastify: FastifyInstance;
```
Comment (frederikpyt) [inline src/modules/hotel/information/__test__/createHotelInformation.test.ts:8]:
```suggestion
import { User, Permission } from "@prisma/client";
import { prisma } from "../../../../plugins/prisma";
import { addTestUserAndPermission } from "../../../../utils/testHelper";

describe("POST /api/hotel/information", () => {
    const fastify = global.fastify;
```

File: src/modules/hotel/information/__test__/createHotelInformation.test.ts [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+    let accessToken: string;
+    let accessTokenNoPermission: string
+
+    beforeAll(async () => {
+        fastify = await build();
+    });
```
Comment (frederikpyt) [inline src/modules/hotel/information/__test__/createHotelInformation.test.ts:14]:
```suggestion
```

File: src/modules/hotel/information/__test__/createHotelInformation.test.ts [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+                name: "Santa Marina Hotel",
+                description: "Santa Marina Hotel is located close to the beach",
+                address: "8130 Sv. Marina, Sozopol, Bulgarien",
+            },
+        });
+    });
+
+    afterAll(async () => {
+        await fastify.close();
+    });
```
Comment (frederikpyt) [inline src/modules/hotel/information/__test__/createHotelInformation.test.ts:33]:
```suggestion
```

File: src/modules/hotel/information/__test__/deleteHotelInformation.test.ts [hunk 8]
```diff
@@ -0,0 +1,110 @@
+import { FastifyInstance } from "fastify";
+import { build } from "../../../../index";
+import { prisma } from "../../../../plugins/prisma";
+import { addTestUserAndPermission } from "../../../../utils/testHelper";
+
+describe("DELETE /api/hotel/information/:id", () => {
+    let fastify: FastifyInstance;
```
Comment (frederikpyt) [inline src/modules/hotel/information/__test__/deleteHotelInformation.test.ts:7]:
```suggestion
import { prisma } from "../../../../plugins/prisma";
import { addTestUserAndPermission } from "../../../../utils/testHelper";

describe("DELETE /api/hotel/information/:id", () => {
    const fastify = global.fastify;
```

File: src/modules/hotel/information/__test__/deleteHotelInformation.test.ts [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+    let accessToken: string;
+    let accessTokenNoPermission: string
+
+    beforeAll(async () => {
+        fastify = await build();
+    });
```
Comment (frederikpyt) [inline src/modules/hotel/information/__test__/deleteHotelInformation.test.ts:13]:
```suggestion
```

File: src/modules/hotel/information/__test__/deleteHotelInformation.test.ts [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+                key: 'Opening Hours',
+                value: '06:00 - 24:00',
+                hotelId: 1000,
+            },
+        });
+    });
+
+    afterAll(async () => {
+        await fastify.close();
+    });
```
Comment (frederikpyt) [inline src/modules/hotel/information/__test__/deleteHotelInformation.test.ts:40]:
```suggestion
```

File: src/modules/hotel/information/__test__/getAllHotelInformation.test.ts [hunk 11]
```diff
@@ -0,0 +1,89 @@
+import { FastifyInstance } from "fastify";
+import { build } from "../../../../index";
+import { prisma } from "../../../../plugins/prisma";
+
+describe("GET /api/hotel/information", () => {
+    let fastify: FastifyInstance;
+
+    beforeAll(async () => {
+        fastify = await build();
+    });
```
Comment (frederikpyt) [inline src/modules/hotel/information/__test__/getAllHotelInformation.test.ts:10]:
```suggestion
import { prisma } from "../../../../plugins/prisma";

describe("GET /api/hotel/information", () => {
    const fastify = global.fastify;
```

File: src/modules/hotel/information/__test__/getAllHotelInformation.test.ts [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+                key: 'Wifi Password',
+                value: 'overnites-is-cool',
+                hotelId: 1000,
+            },
+        });
+    });
+
+    afterAll(async () => {
+        await fastify.close();
+    });
```
Comment (frederikpyt) [inline src/modules/hotel/information/__test__/getAllHotelInformation.test.ts:52]:
```suggestion
```

File: src/modules/hotel/information/__test__/getHotelInformation.test.ts [hunk 13]
```diff
@@ -0,0 +1,72 @@
+import { FastifyInstance } from "fastify";
+import { build } from "../../../../index";
+import { prisma } from "../../../../plugins/prisma";
+
+describe("GET /api/hotel/information/:id", () => {
+    let fastify: FastifyInstance;
+
+    beforeAll(async () => {
+        fastify = await build();
+    });
```
Comment (frederikpyt) [inline src/modules/hotel/information/__test__/getHotelInformation.test.ts:10]:
```suggestion
import { prisma } from "../../../../plugins/prisma";

describe("GET /api/hotel/information/:id", () => {
    const fastify = global.fastify;
```

File: src/modules/hotel/information/__test__/getHotelInformation.test.ts [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+                key: 'Opening Hours',
+                value: '06:00 - 24:00',
+                hotelId: 1000,
+            },
+        });
+    });
+
+    afterAll(async () => {
+        await fastify.close();
+    });
```
Comment (frederikpyt) [inline src/modules/hotel/information/__test__/getHotelInformation.test.ts:36]:
```suggestion
```

File: src/modules/hotel/information/__test__/updateHotelInformation.test.ts [hunk 15]
```diff
@@ -0,0 +1,220 @@
...
+import { addTestUserAndPermission } from "../../../../utils/testHelper";
+
+describe("PUT /api/hotel/information", () => {
+    let fastify: FastifyInstance;
+    let accessToken: string;
+    let accessTokenNoPermission: string
+
+    beforeAll(async () => {
+        fastify = await build();
+    });
```
Comment (frederikpyt) [inline src/modules/hotel/information/__test__/updateHotelInformation.test.ts:13]:
```suggestion
import { prisma } from "../../../../plugins/prisma";
import { addTestUserAndPermission } from "../../../../utils/testHelper";

describe("PUT /api/hotel/information", () => {
    const fastify = global.fastify;
    let accessToken: string;
    let accessTokenNoPermission: string
```

File: src/modules/hotel/information/__test__/updateHotelInformation.test.ts [hunk 16]
```diff
@@ -0,0 +1,220 @@
...
+                key: 'Opening Hours',
+                value: '06:00 - 24:00',
+                hotelId: 1000,
+            },
+        });
+    });
+
+    afterAll(async () => {
+        await fastify.close();
+    });
```
Comment (frederikpyt) [inline src/modules/hotel/information/__test__/updateHotelInformation.test.ts:40]:
```suggestion
```

File: src/modules/hotel/information/information.schema.ts [hunk 17]
```diff
@@ -0,0 +1,80 @@
...
+        invalid_type_error: "Number must be an int",
+    }),
+    hotelId: z.number({
+        required_error: "Hotel ID is required",
+        invalid_type_error: "Hotel ID must be an int",
+    }),
+};
+
+const getHotelInformationSchema = z.object({
+    id: z.string(),
```
Comment (frederikpyt) [inline src/modules/hotel/information/information.schema.ts:20]:
```suggestion
    id: z.number(),
```

**Review by frederikpyt — CHANGES_REQUESTED** [review]
For the tests to work after the suggested changes you may need to merge with master

**Review by frederikpyt — APPROVED** [review]

**Pull request merged**